File: LeetCode/GO/left_right_sum_difference.go
// https://leetcode.com/problems/left-and-right-sum-differences/description/

package main

import "fmt"

func leftRightDifference(nums []int) []int {
	lSum, rSum := 0, 0
	n := len(nums)
	answer := make([]int, n)

	for _, val := range nums {
		rSum += val
	}

	for pos, val := range nums {

		if pos > 0 {
			lSum += nums[pos-1]
		}
		rSum -= val

		answer[pos] = abs(lSum - rSum)
	}
	return answer
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

func main() {

	nums1 := []int{10, 4, 8, 3}
	nums2 := []int{1}

	fmt.Printf("The ans for %v is %v \n", nums1, leftRightDifference(nums1))
	fmt.Printf("The ans for %v is %v \n", nums2, leftRightDifference(nums2))

}

File: LeetCode/Python/left_right_sum_difference.py
"""
https://leetcode.com/problems/left-and-right-sum-differences/description/
"""

from typing import List

class Solution:
    def leftRightDifference(self, nums: List[int]) -> List[int]:
        

        l_sum , r_sum = 0, sum(nums)

        n = len(nums)

        answer = [0] * n

        for pos, val in enumerate(nums):

            l_sum += nums[pos-1] if pos > 0 else 0

            r_sum -= nums[pos]

            answer[pos] = abs(l_sum - r_sum)
        
        return answer



if __name__ == "__main__":
    
    nums = [10,4,8,3]
    nums2 = [1]
    
    s  = Solution()    
    print(f" ans for {nums} is   {s.leftRightDifference(nums=nums)}")
    print(f" ans for {nums2}   {s.leftRightDifference(nums=nums2)}")

